homeassistant/components/device_tracker/__init__.py

```diff
@@ -236,10 +238,51 @@ def see(hass: HomeAssistantType, mac: str = None, dev_id: str = None,
 
 async def async_setup_entry(hass, entry):
     """Set up an entry."""
-    await hass.data[DOMAIN](entry.domain, entry)
+    await hass.data[DOMAIN].get_async_setup_platform()(entry.domain, entry)
     return True
 
 
+async def async_unload_entry(hass, entry):
+    """Unload an entry."""
+    platform = hass.data[DOMAIN].get_platform(entry.domain)
+    if platform and hasattr(platform, 'async_unload_entry'):
+        await platform.async_unload_entry(hass, entry)
+        return True
+    return False
+
+
+class PlatformManager:
+    """Store data needed to unload entries from loaded platforms."""
+
+    def __init__(self, async_setup_platform: callable):
+        """Initialize a platform manager."""
+        self.platforms = {}
+        self.async_setup_platform = async_setup_platform
+
+    def add_platform(self, platform_name: str, platform):
+        """Add a platform that has been loaded to the platform manager."""
+        _LOGGER.debug('Adding %s to PlatformManager', platform_name)
+        if platform_name is None:
+            return
+        if platform_name in self.platforms:
```

Comment:
If we've added a platform, but the platform setup fails, we would need to remove the platform to not trigger this warning when setting up the platform again.

Reply:
Yes, definitely. Added support for that.